processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -48,12 +62,57 @@
   private final ConcurrentMap<String, LookupExtractorFactory> lookupMap = new ConcurrentHashMap();
   private final Object lock = new Object();
   private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ListeningScheduledExecutorService executor = MoreExecutors.listeningDecorator(Executors.newScheduledThreadPool(
+      1,
+      Execs.makeThreadFactory("LookupRefManager--%s")
+  ));
+  private final LookupConfig lookupConfig;
+  private final LookupSnapshot lookupSnapshot;
+
+  @Inject
+  public LookupReferencesManager(LookupConfig lookupConfig, final @Json ObjectMapper objectMapper)
+  {
+    this.lookupConfig = lookupConfig;
+    if (Strings.isNullOrEmpty(lookupConfig.getSnapshotWorkingDir())) {
+      this.lookupSnapshot = null;
+    } else {
+      this.lookupSnapshot = new LookupSnapshot(objectMapper, lookupConfig.getSnapshotWorkingDir());
+    }
+  }
 
   @LifecycleStart
   public void start()
   {
     synchronized (lock) {
       if (!started.getAndSet(true)) {
+        if (lookupSnapshot != null) {
+          final List<LookupBean> lookupBeanList = lookupSnapshot.pullExistingSnapshot();
+          for (LookupBean lookupBean : lookupBeanList
+              ) {
+            this.put(lookupBean.name, lookupBean.factory);
+          }
+          if (lookupConfig.getSnapshotPeriod().getMillis() != 0) {
```

Comment:
I don't see the benefit in taking periodic snapshots.  This exact same class has the mutation methods that tell us when the state has changed.  Why are we adding yet another background process just to persist something that doesn't change very often and we have a very easy and clear way to know when it does change?